pkg/reconciler/nodenetworkconfig_reconciler.go

```diff
@@ -0,0 +1,274 @@
+	reconciler := &NodeNetworkConfigReconciler{
+		client:         clusterClient,
+		netlinkManager: netlinkManager,
+		frrManager:     frrManager,
+		anycastTracker: anycastTracker,
+		logger:         logger,
+		nodeConfigPath: nodeConfigPath,
+	}
+
+	cfg, err := config.LoadConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error loading config: %w", err)
+	}
+	reconciler.config = cfg
+
+	if val := os.Getenv("FRR_CONFIG_FILE"); val != "" {
+		reconciler.frrManager.SetConfigPath(val)
+	}
+
+	if err := reconciler.frrManager.Init(cfg.SkipVRFConfig[0]); err != nil {
+		return nil, fmt.Errorf("error trying to init FRR Manager: %w", err)
+	}
+
+	nc, err := healthcheck.LoadConfig(healthcheck.NetHealthcheckFile)
+	if err != nil {
+		return nil, fmt.Errorf("error loading networking healthcheck config: %w", err)
+	}
+
+	tcpDialer := healthcheck.NewTCPDialer(nc.Timeout)
+	reconciler.healthChecker, err = healthcheck.NewHealthChecker(reconciler.client,
+		healthcheck.NewDefaultHealthcheckToolkit(reconciler.frrManager, tcpDialer),
+		nc)
+	if err != nil {
+		return nil, fmt.Errorf("error creating networking healthchecker: %w", err)
+	}
+
+	reconciler.nodeConfig, err = readNodeConfig(reconciler.nodeConfigPath)
+	if !errors.Is(err, os.ErrNotExist) {
+		return nil, fmt.Errorf("error reading NodeConfig from disk: %w", err)
+	}
+
+	return reconciler, nil
+}
+
+func (reconciler *NodeNetworkConfigReconciler) Reconcile(ctx context.Context) error {
+	r := &reconcileNodeNetworkConfig{
+		NodeNetworkConfigReconciler: reconciler,
+		Logger:                      reconciler.logger,
+	}
+
+	if err := r.config.ReloadConfig(); err != nil {
+		return fmt.Errorf("error reloading network-operator config: %w", err)
+	}
+
+	// get NodeConfig from apiserver
+	cfg, err := r.fetchNodeConfig(ctx)
+	if err != nil {
+		// discard IsNotFound error
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	if r.nodeConfig != nil && r.nodeConfig.Spec.Revision == cfg.Spec.Revision {
+		// current in-memory conifg has the same revision as the fetched one
+		// this means that config was already provisioned - skip
+		if cfg.Status.ConfigStatus != StatusProvisioned {
+			if err := setStatus(ctx, r.client, cfg, StatusProvisioned); err != nil {
+				return fmt.Errorf("error setting config status: %w", err)
+			}
+		}
+		return nil
+	}
+
+	// config is invalid - discard
+	if cfg.Status.ConfigStatus == StatusInvalid {
+		return nil
+	}
+	if err := r.processConfig(ctx, cfg); err != nil {
+		return fmt.Errorf("error while processing config: %w", err)
+	}
+
+	// replace in-memory working config and store it on the disk
+	reconciler.nodeConfig = cfg
+	if err := storeNodeConfig(cfg, reconciler.nodeConfigPath); err != nil {
+		return fmt.Errorf("error saving NodeConfig status: %w", err)
```

Comment:
if this fails, we'll never attempt to store it again, since the reconciliation is aborted in the if statement at L113. Does it make sense to just store it to disk during each reconciliation when the in-memory revision matches the reconciled revision? Otherwise we could end up with weird state in case we ever load outdated config from disk and at the same time create a new revision (admittedly rather rare, but could happen when we roll out a new version and change config at the same time). If that new revision is invalid we would then roll back to an outdated version, which is unexpected.

Reply:
I've added additional config storing  inside the if you've mentioned (L113).